mne/preprocessing/ica.py

```diff
@@ -221,6 +221,10 @@ def __init__(self, n_components=None, max_pca_components=None,
 
         if fit_params is None:
             fit_params = {}
+        fit_params = deepcopy(fit_params)  # avoid side effects
+        if "extended" in fit_params:
```

Comment:
I think it should be
```
if method == "infomax" and fit_params.get("extended", False):
```

Because in principle, somebody might do `method='infomax', fit_params={"extended":False}`.